c/src/fetch_asset.rs

```diff
@@ -0,0 +1,96 @@
+use std::ffi::c_void;
+
+use foxglove::websocket::{AssetHandler, AssetResponder};
+
+use crate::{bytes::FoxgloveBytes, FoxgloveString};
+
+pub struct FoxgloveFetchAssetResponder(AssetResponder);
+impl FoxgloveFetchAssetResponder {
+    /// Moves the responder to the heap and returns a raw pointer.
+    ///
+    /// After calling this function, the caller is responsible for eventually calling
+    /// [`Self::from_raw`] to recover the responder.
+    fn into_raw(self) -> *mut Self {
+        Box::into_raw(Box::new(self))
+    }
+
+    /// Recovers the boxed responder from a raw pointer.
+    ///
+    /// # Safety
+    /// - The raw pointer must have been obtained from [`Self::into_raw`]
+    unsafe fn from_raw(ptr: *mut Self) -> Box<Self> {
+        unsafe { Box::from_raw(ptr) }
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct FetchAssetHandler {
+    context: *const c_void,
```

Comment:
Same comment about the name, if you rename it in the previous PR don't forget this one as well